### ISSUE TITLE
Defining J9SW_JIT_FLOATS_PASSED_AS_DOUBLES on Power

### ISSUE THREAD

**Issue by 0xdaryl** [eclipse/openj9#6369]
Why is Power the only platform to define the stackwalker macro `J9SW_JIT_FLOATS_PASSED_AS_DOUBLES`?   I understand the effect in the stackwalker when it's defined (or at least I think I do), but it's not clear to me what the Power JIT is doing that's different from the other architectures that requires it to be defined.

Any insight is appreciated.  It will probably turn out to be a "duh!" moment for me.  :-)

@zl-wang @gita-omr @gacholio

**Comment by zl-wang** [comment]
Historically, there is no single-precision format representation in register on POWER architecture.  It is still true as to the scalar part of the architecture.  SP float is only available in memory.  SP value is represented in DP format in register. There is rounding to SP instruction, but that just means preparing for SP values (format is still DP).

I guessed this is the reason for that flag.

**Comment by gacholio** [comment]
Yes, it's because P doesn't split the double registers into two float registers. It may also be desireable to do this for performance reasons on other architectures.

**Comment by zl-wang** [comment]
Overall, this is a transparent scheme of hardware implementation.  By and large, you don't need to know SP value is represented in DP format in register, as long as you stick to type promotion rules of high-level language.  So, one performance side-effect:  if you feed a true DP value into a single-precision operation without rounding it to SP first, you may cause performance glitches on certain hardware.  For example, fadds fr1, fr2, fr3   (single-precision add:  fr1 = fr2 + fr3)  ... if one of fr2 or fr3 is DP value (not rounded to SP first, assuming it came from a prior DP operation), it might have performance glitches (even undefined side-effect) ... I guessed this is largely true on other architectures.
